### PR TITLE
Analysis update: add DMG, EGFR to methyl mol subtype

### PR DESCRIPTION
**Pull request by zzgeng**
#### What scientific question is your analysis addressing?
Add DMG, EGFR to methyl mol subtype

#### What was your approach?
Add DMG, EGFR to methyl mol subtype but maintain the previous molecular subtype results as DMG, H3 K28

#### What GitHub issue does your pull request address?
https://github.com/d3b-center/OpenPedCan-analysis/issues/448

#### Which areas should receive a particularly close look?
HGG module output and histologies.tsv

#### Is the analysis in a mature enough form that the resulting figure(s) and/or table(s) are ready for review?
Yes

#### What types of results are included (e.g., table, figure)?
Table

#### Reproducibility Checklist

- [x] The dependencies required to run the code in this pull request have been added to the project Dockerfile.
- [x] This analysis has been added to continuous integration.

#### Documentation Checklist

- [x] This analysis module has a `README` and it is up to date.
- [x] This analysis is recorded in the table in `analyses/README.md` and the entry is up to date.
- [x] The analytical code is documented and contains comments.

### CODE REVIEW

File: analyses/molecular-subtyping-HGG/results/HGG_cleaned_all_table.tsv [hunk 1]
```diff
@@ -795,7 +795,7 @@ PT_BC0B16HC	BS_PYYNXW86	BS_1D93Z575	BS_ENH6KT2R	7316-7075	7316-7075_Solid Tissue
...
 PT_7S5BVPE2	NA	NA	BS_548VV6P6	7316-712	7316-712_Solid Tissue	NA	NA	NA	NA	NA	NA	NA	NA	NA	NA	NA	NA	NA	NA	NA	NA	NA	NA	NA	NA	NA	NA	NA	NA	NA	NA	Solid Tissue	NA	NA	NA	NA	NA	NA	NA	NA	NA	NA	NA	NA	DMG, H3 K28	DMG, H3 K28
-PT_FCWMT41S	BS_XK482X70	NA	BS_4Q7PQHS8	7316-7133	7316-7133_Solid Tissue	Midline	5	No	No	No	No	No	No	No	No	No	No	None	No R132 or R172	None	No V600E	neutral	neutral	neutral	neutral	neutral	neutral	neutral	neutral	neutral	neutral	Solid Tissue	NA	NA	NA	NA	NA	NA	NA	NA	NA	NA	NA	NA	DMG, H3 K28	DMG, H3 K28
+PT_FCWMT41S	BS_XK482X70	NA	BS_4Q7PQHS8	7316-7133	7316-7133_Solid Tissue	Midline	5	No	No	No	No	No	No	No	No	No	No	None	No R132 or R172	None	No V600E	neutral	neutral	neutral	neutral	neutral	neutral	neutral	neutral	neutral	neutral	Solid Tissue	NA	NA	NA	NA	NA	NA	NA	NA	NA	NA	NA	NA	DMG, EGFR	HGG, H3 wildtype
```
Comment (jharenza) [inline analyses/molecular-subtyping-HGG/results/HGG_cleaned_all_table.tsv]: you might need another rule that if there is no mutation, take DMG, EGFR and make it DMG, K28 as the molecular subtype. I see a few being converted back to wildtype, but we should just include this when we do that final search using methyl only.

Comment (zzgeng) [reply analyses/molecular-subtyping-HGG/results/HGG_cleaned_all_table.tsv]: Thank you for pointing out. This issue is addressed.

File: analyses/molecular-subtyping-pathology/results/compiled_molecular_subtypes_with_clinical_pathology_feedback_and_report_info.tsv [hunk 2]
```diff
@@ -4669,8 +4669,8 @@ PT_FC0EF5ZT	7316-3572	BS_WJ7GMQTF	Low-grade astrocytic tumor	Ganglioglioma	Gangl
 PT_FCM7XPVR	7316-2285	BS_4GKH983E	Ependymal tumor	Ependymoma	NA	EPN, To be classified	NA	Recurrence	NA	NA
 PT_FCM7XPVR	7316-2285	BS_554AS76R	Ependymal tumor	Ependymoma	NA	EPN, To be classified	NA	Recurrence	NA	NA
 PT_FCM7XPVR	7316-2285	BS_MQCKXD60	Ependymal tumor	Ependymoma	NA	EPN, To be classified	NA	Recurrence	NA	NA
-PT_FCWMT41S	7316-7133	BS_4Q7PQHS8	Diffuse astrocytic and oligodendroglial tumor	HGAT	NA	DMG, EGFR	NA	Initial CNS Tumor	NA	DMG, EGFR
-PT_FCWMT41S	7316-7133	BS_XK482X70	Diffuse astrocytic and oligodendroglial tumor	HGAT	NA	DMG, EGFR	NA	Initial CNS Tumor	NA	DMG, EGFR
+PT_FCWMT41S	7316-7133	BS_4Q7PQHS8	Diffuse astrocytic and oligodendroglial tumor	HGAT	Diffuse midline glioma, H3 K28-mutant	DMG, H3 K28	Updated via OpenPedCan subtyping	Initial CNS Tumor	NA	DMG, EGFR
```
Comment (jharenza) [inline analyses/molecular-subtyping-pathology/results/compiled_molecular_subtypes_with_clinical_pathology_feedback_and_report_info.tsv:4672]: oh gosh - I thought we had updated this in v12, @zzgeng can you use "Diffuse midline glioma, H3 K28-altered" here, not "mutant" for all of these samples?

**Review by AntoniaChroni — APPROVED** [review]
I run the all bash files on EC2 and it run with no problems. All relative files were generated with integrated changes.

- `plots/HGG_stranded.pdf`: it seems there is no grey/ no `HGAT_NO_HISTONE_BRAF`, is that right?

**Review by jharenza — APPROVED** [review]

**Pull request merged**